components/query/partials/feature-popup.html

```diff
@@ -1,47 +1,43 @@
-<div ng-style="popupVisible()" class="hs-hover-popup">
+<div [ngStyle]="popupVisible()" class="hs-hover-popup">
     <div class="card" style="padding: 5px;">
-        <div ng-repeat="layerDesc in queryBaseService.featureLayersUnderMouse" style="line-height: 1.2">
+        <div *ngFor="let layerDesc of HsQueryBaseService.featureLayersUnderMouse" style="line-height: 1.2">
             <strong style="font-size: small;">{{layerDesc.title}}</strong><br>
-            <div ng-repeat="feature in layerDesc.features">
-                <div ng-show="isClustered(feature)" style="font-size: small; line-height: 1; margin-bottom: 2px; border-bottom: 1px solid rgb(199, 199, 199)">
-                    <div ng-repeat="subFeature in feature.get('features')" ng-show="subFeature.attributesForHover.length > 0">
+            <div *ngFor="let feature of layerDesc.features">
+                <div [hidden]="!isClustered(feature)" style="font-size: small; line-height: 1; margin-bottom: 2px; border-bottom: 1px solid rgb(199, 199, 199)">
+                    <div *ngFor="let subFeature of feature.get('features')" [hidden]="!(subFeature.attributesForHover.length > 0)">
                         {{serializeFeatureName(subFeature)}}
-                        <button class="btn btn-danger float-right btn-sm" ng-if="isFeatureRemovable(subFeature)" ng-click="removeFeature(subFeature)"
-                            translate data-toggle="tooltip" title="{{'Remove feature' | translate}}">
+                        <button class="btn btn-danger float-right btn-sm" *ngIf="isFeatureRemovable(subFeature)" (click)="removeFeature(subFeature)" i18n data-toggle="tooltip" i18n-title="Remove feature">
                             <i class="icon-remove-circle"></i>
                         </button>
-                        <div ng-repeat="attribute in subFeature.attributesForHover">
+                        <div *ngFor="let attribute of subFeature.attributesForHover">
                             <div class="d-flex flex-row">
                                 <div class="p-1 flex-grow-1">
-                                    {{attribute.key}}&emsp;
```

Comment:
I believe the `&emsp;` entity was quite intentional here. Otherwise the key and the value could be rendered as `keyvalue`.

Reply:
EN space returned

Reply:
Strange, but I can see it as missing still (replaced by regular space)...